lib/borda_poll.rb

```diff
@@ -0,0 +1,46 @@
+require 'set'
+
+class BordaPoll
+  class InvalidCandidateError < StandardError
+  end
+  class MultipleVoteError < StandardError
+  end
+  class OverdueVoteError < StandardError
+  end
+
+  attr_reader :title, :candidates, :votes, :closing
+
+  def initialize(title, candidates, closing)
+    @title = title
+    @candidates = candidates
+    @closing = closing
+    @votes = []
+  end
+
+  def add_vote(vote)
+    unless (vote.candidates.to_set ^ @candidates.to_set).length == 0 then
+      raise InvalidCandidateError.new(vote.candidates)
+    end
+
+    if @votes.map { |element| element.voter }.include?(vote.voter) then
```

Comment:
[nit]

- `arr.map { |elm| elm.foo }` は `arr(&:foo)` と書くのが ruby っぽいです
  - 詳しくは https://qiita.com/k-penguin-sato/items/420d7487b28b5d58cac4 など
- `then` は省略可能です（他も同様）

```suggestion
    if @votes.map(&:voter).include?(vote.voter)
```

※ [nit] は nitpick の略で「細かい指摘なのでどちらでもいいけど」みたいなニュアンスで使ってます。

Reply:
どちらも知らなかったです。ありがとうございます。